testmain.go

```diff
@@ -60,10 +60,26 @@ func VerifyTestMain(m TestingM, options ...Option) {
 	}
 	defer func() { cleanup(exitCode) }()
 
-	if exitCode == 0 {
+	var (
+		run      bool
+		errorMsg string
+	)
+
+	if !opts.runOnFailure && exitCode == 0 {
```

Comment:
I don't think `opts.runOnFailure` should matter here.
to keep the old behaviour, we only need to check on `exitCode`.

```
	if exitCode == 0 {
		errorMsg = "goleak: Errors on successful test run:%v\n"
		run = true
	} else if opts.runOnFailure {
		errorMsg = "goleak: Errors on unsuccessful test run: %v\n"
		run = true
	}
```

Reply:
I opened https://github.com/uber-go/goleak/pull/132 in case this went unnoticed